### PR TITLE
[SRVOCF-477] add default s2i builders.

### PR DESCRIPTION
**Pull request by Kaustubh-pande**
as per title
cc- @dsimansk @jrangelramos @lance

### CODE REVIEW

File: hack/build.sh.d/func.sh [hunk 1]
```diff
@@ -2,6 +2,13 @@
 
 # There no midstream image currently, fallback to upstream one.
 readonly func_util_image="${KN_PLUGIN_FUNC_UTIL_IMAGE:-quay.io/boson/alpine-socat:1.7.4.3-r1-non-root}"
+readonly default_node_builder_image="${DEFAULT_NODE_BUILDER_IMAGE:-registry.access.redhat.com/ubi8/nodejs-16}"
+readonly default_quarkus_builder_image="${DEFAULT_QUARKUS_BUILDER_IMAGE:-registry.access.redhat.com/ubi8/openjdk-17}"
+readonly default_python_builder_image="${DEFAULT_PYTHON_BUILDER_IMAGE:-registry.access.redhat.com/ubi8/python-39}"
+
 export EXTERNAL_LD_FLAGS="${EXTERNAL_LD_FLAGS:-} \
 -X knative.dev/func/pkg/k8s.SocatImage=${func_util_image} \
--X knative.dev/func/pkg/k8s.TarImage=${func_util_image}"
```
[the remark this answers is not in the document]
Comment (Kaustubh-pande) [reply hack/build.sh.d/func.sh:7]: Not removed replaced below.

[the remark this answers is not in the document]
Comment (lance) [reply hack/build.sh.d/func.sh:7]: 👍 Sorry I missed that!

**Review by lance — APPROVED** [review]
/lgtm

File: hack/build.sh.d/func.sh [hunk 2]
```diff
@@ -2,6 +2,13 @@
...
\ No newline at end of file
+-X knative.dev/func/pkg/k8s.TarImage=${func_util_image} \
+-X knative.dev/func/pkg/builders/s2i/s2i.DefaultNodeBuilder=${default_node_builder_image} \
+-X knative.dev/func/pkg/builders/s2i/s2i.DefaultQuarkusBuilder=${default_quarkus_builder_image} \
+-X knative.dev/func/pkg/builders/s2i/s2i.DefaultPythonBuilder=${default_python_builder_image}"
```
Comment (jrangelramos) [inline hack/build.sh.d/func.sh:14]: Here it needs to remove the extra path `s2i/` to make it work properly
 
```suggestion
-X knative.dev/func/pkg/builders/s2i.DefaultNodeBuilder=${default_node_builder_image} \
-X knative.dev/func/pkg/builders/s2i.DefaultQuarkusBuilder=${default_quarkus_builder_image} \
-X knative.dev/func/pkg/builders/s2i.DefaultPythonBuilder=${default_python_builder_image}"
```

**Review by jrangelramos — COMMENTED** [review]
Hello @Kaustubh-pande we need that extra change otherwise it won't catch the var.

File: hack/build.sh.d/func.sh (hunk 2 above)
Comment (Kaustubh-pande) [reply hack/build.sh.d/func.sh:14]: Thanks! Good catch!

**Pull request merged**